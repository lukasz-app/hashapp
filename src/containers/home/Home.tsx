import React, { Component } from 'react';
import { View, TextInput } from 'react-native';
import styles from './styles';
import { HashView, Button } from '../../components';
import SHA1 from '../../util/sha1';

interface Props {}
interface State {
  text: string;
  textHashed: string;
  hash: string;
}

export default class Hello extends Component<Props, State> {
  state = {
    text: '',
    hash: '',
    textHashed: ''
  };

  hashInput = () => {
    this.setState({ hash: SHA1(this.state.text), textHashed: this.state.text });
  };

  onChangeText = (text: string) => {
    this.setState({ text });
  };

  render() {
    const { hash, text, textHashed } = this.state;
    const hashViewLabel = `SHA1 hash for "${textHashed}"`;
    return (
      <View style={styles.container}>
        <View style={styles.inputRowContainer}>
          <View style={styles.inputContainer}>
            <TextInput
              placeholder="Enter text to be hashed"
              autoFocus
              multiline
              onChangeText={this.onChangeText}
            />
          </View>
          <Button label="Hash" onPress={this.hashInput} disabled={!text.length} />
        </View>
        {!!hash.length && <HashView hash={hash} label={hashViewLabel} />}
      </View>
    );
  }
}
