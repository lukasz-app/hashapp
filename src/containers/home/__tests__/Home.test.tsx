import React from 'react';
import renderer from 'react-test-renderer';

import Home from '../Home';

it('renders correctly', () => {
  const button = renderer.create(<Home />).toJSON();
  expect(button).toMatchSnapshot();
});
