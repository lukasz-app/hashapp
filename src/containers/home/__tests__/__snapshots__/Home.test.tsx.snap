// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "stretch",
      "backgroundColor": "#FBFCFE",
      "flex": 1,
      "paddingBottom": 20,
      "paddingHorizontal": 20,
      "paddingTop": 30,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "borderColor": "#6B6E72",
          "borderRadius": 3,
          "borderWidth": 1,
          "flex": 1,
          "marginRight": 6,
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        autoFocus={true}
        multiline={true}
        numberOfLines={2}
        onChangeText={[Function]}
        placeholder="Enter text to be hashed"
        returnKeyType="done"
        underlineColorAndroid="transparent"
      />
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#DDDDDD",
          "borderRadius": 3,
          "opacity": 1,
          "paddingHorizontal": 16,
          "paddingVertical": 8,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#FBFCFE",
              "fontSize": 15,
              "fontWeight": "600",
            },
          ]
        }
      >
        Hash
      </Text>
    </View>
  </View>
</View>
`;
